api/app/pagination.py

```diff
@@ -1,7 +1,61 @@
+import base64
+import json
+from collections import OrderedDict
+
+from flag_engine.identities.builders import build_identity_model
 from rest_framework.pagination import PageNumberPagination
+from rest_framework.response import Response
+from rest_framework.utils.urls import replace_query_param
 
 
 class CustomPagination(PageNumberPagination):
     page_size = 999
     page_size_query_param = "page_size"
     max_page_size = 999
+
+
+class EdgeIdentityPagination(CustomPagination):
+    def paginate_queryset(self, dynamo_queryset, request, view=None):
+        self.previous_last_evaluated_key = request.GET.get("last_evaluated_key")
+        self.request = request
+        last_evaluated_key = dynamo_queryset.get("LastEvaluatedKey")
+        if last_evaluated_key:
+            self.last_evaluated_key = base64.b64encode(
+                json.dumps(last_evaluated_key).encode()
+            )
+
+        return [
```

Comment:
I don't feel very good about converting dynamo document to dict here what do you think?

Reply:
>I don't feel very good about converting dynamo document to dict
I think you mean dynamo document to engine model? 

>but if we do that on the get_queryset method than we have to override list to pass last_evaluated_key to this
I don't really follow this statement. Can you illustrate your point with a code snippet perhaps?

Reply:
>I think you mean dynamo document to engine model?

yeah that is correct 

>I don't really follow this statement. Can you illustrate your point with a code snippet perhaps?

yeah, so currently we are using the default implementation of `list`, because `get_queryset` returns the raw response of `dynamodb query api` instead of returning a list of engine model instances. But if we update `get_queryset` to return a list of engine model instance instead of the raw dynamodb response, we will have to override list to look something like

```
    def list(self, request, *args, **kwargs):
        queryset = self.filter_queryset(self.get_queryset())
        last_evaluated_key = some_method()
        page = self.paginate_queryset(queryset, last_evaluated_key)
        if page is not None:
            serializer = self.get_serializer(page, many=True)
            return self.get_paginated_response(serializer.data)

        serializer = self.get_serializer(queryset, many=True)
        return Response(serializer.data)



```

Reply:
Ok, understood. Then yeah, I definitely prefer this approach.